appstudio-controller/controllers/appstudio.redhat.com/environment_controller.go

```diff
@@ -139,41 +140,82 @@ func (r *EnvironmentReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 
 	}
 
+	// Reconcile the Environment into a ManagedEnvironment, based on Environment CR contents
+	// - 'res' and 'err' work as normal for a Reconcile function
+	// - 'statusUpdated' is true if an error occurred during reconciliation, which caused the .status.Conditions[EnvironmentConditionErrorOccurred] field to be set. It is false otherwise.
+	res, err, statusUpdated := reconcileEnvironment(ctx, *environment, rClient, log)
```

Comment:
After fixing the original issue, I was seeing cases where the Environment reconcile was being called over and over in an infinite loop.

This was due to the .status.Conditions of Environment being set repeatedly: in multiple places in the old code, we would reset Environment's .status.conditions to indicate that ErrorOccurred was resolved.  

I've instead updating this file to (ideally) only reset ErrorOccurred to resolved once: functions within this controller can now return true/false to indicate whether they updated ErrorOccurred due to an error.
- If this value is false, we remove ErrorOccured.